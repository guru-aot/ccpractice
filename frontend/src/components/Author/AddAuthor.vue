
/**
 * AuthorComponent
 */
 <template>
  <v-card class="mx-auto" style="max-width: 500px;">
    <v-alert type="success" v-if="successStatus">Author Added succesfully</v-alert>
    <v-alert type="error" v-if="errorStatus">Something went wrong...</v-alert>

    <v-toolbar color="deep-purple accent-4" cards dark flat>
      <v-card-title class="title font-weight-regular">Add Author</v-card-title>
      <v-spacer></v-spacer>
    </v-toolbar>
    <v-form ref="form" v-model="form" class="pa-4 pt-6">
      <v-text-field
        v-model="firstName"
        filled
        color="deep-purple"
        counter="10"
        label="First name"
        style="min-height: 96px"
        type="text"
        :rules="[rules.required]"
      ></v-text-field>
      <v-text-field
        v-model="lastName"
        filled
        color="deep-purple"
        counter="10"
        label="Last name"
        style="min-height: 96px"
        type="text"
        :rules="[rules.required]"
      ></v-text-field>
      <v-text-field
        v-model="middleName"
        filled
        color="deep-purple"
        counter="10"
        label="Middle name"
        style="min-height: 96px"
        type="text"
        :rules="[rules.required]"
      ></v-text-field>
    </v-form>
    <v-divider></v-divider>
    <v-card-actions>
      <v-btn text @click="$refs.form.reset()">Clear</v-btn>
      <v-spacer></v-spacer>
      <v-btn
        :disabled="!form"
        :loading="isLoading"
        class="white--text"
        color="deep-purple accent-4"
        depressed
        @click="addAuthor"
      >Submit</v-btn>
    </v-card-actions>
  </v-card>
</template>
 <script lang="ts">
import { Component, Vue } from 'vue-property-decorator';
import { Getter, namespace, Action } from 'vuex-class';
const AuthorModule = namespace('AuthorModule');

@Component
export default class AuthorComponent extends Vue {
  @AuthorModule.Getter('successStatus') public successStatus!: boolean;
  @AuthorModule.Getter('errorStatus') public errorStatus!: boolean;
  @AuthorModule.Action('addAuthor') public addAuthorStore!: any;
  public form: boolean = false;
  private isLoading: boolean = false;
  private firstName: string = '';
  private lastName: string = '';
  private middleName: string = '';

  private rules = {
    length: (len: any) => (v: any) =>
      (v || '').length >= len || 'Invalid character length, required' + len,

    required: (v: any) => !!v || 'This field is required'
  };
  /**
   * @description
   */
  private addAuthor() {
    const data = {
      first_name: this.firstName,
      last_name: this.lastName,
      middle_name: this.middleName
    };

    this.addAuthorStore(data);
    // this.$refs.form.reset();
    (this.$refs.form as HTMLFormElement).reset();
  }
}
</script>
 