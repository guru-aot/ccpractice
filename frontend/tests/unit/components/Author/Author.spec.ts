/**
 * AddAuthor test
 */
import { shallowMount, createLocalVue } from '@vue/test-utils';
import AddAuthor from '@/components/Author/AddAuthor.vue';

import Vuex from 'vuex';
import vuetify from 'vuetify';

describe('component/Author/AddAuthor.vue', () => {
  let wrapper: any;
  beforeEach(() => {
    const localVue = createLocalVue();
    localVue.use(Vuex);
    localVue.use(vuetify);
    const store = new Vuex.Store({
      modules: {
        AuthorModule: {
          namespaced: true,
          state: {},
          getters: { successStatus: jest.fn(), errorStatus: jest.fn() },
          actions: {},
        },
      },
    });
    wrapper = shallowMount(AddAuthor, {
      localVue,
      store,
    });
  });
  it('renders props when passed', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
