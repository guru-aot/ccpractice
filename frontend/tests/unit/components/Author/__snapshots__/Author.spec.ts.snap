// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/Author/AddAuthor.vue renders props when passed 1`] = `
<v-card-stub
  class="mx-auto"
  loaderheight="4"
  style="max-width: 500px;"
  tag="div"
>
  <!---->
   
  <!---->
   
  <v-toolbar-stub
    cards=""
    color="deep-purple accent-4"
    dark="true"
    extensionheight="48"
    flat="true"
    src=""
    tag="header"
    tile="true"
  >
    <v-card-title-stub
      class="title font-weight-regular"
    >
      Add Author
    </v-card-title-stub>
     
    <v-spacer-stub />
  </v-toolbar-stub>
   
  <v-form-stub
    class="pa-4 pt-6"
  >
    <v-text-field-stub
      backgroundcolor=""
      clearicon="$clear"
      color="deep-purple"
      counter="10"
      errorcount="1"
      errormessages=""
      filled="true"
      label="First name"
      loaderheight="2"
      messages=""
      rules="v => {
          /* istanbul ignore next */cov_15j8hk4u63.f[4]++;
          cov_15j8hk4u63.s[17]++;
          return (/* istanbul ignore next */(cov_15j8hk4u63.b[2][0]++, !!v) || /* istanbul ignore next */(cov_15j8hk4u63.b[2][1]++, 'This field is required')
          );
        }"
      style="min-height: 96px;"
      successmessages=""
      type="text"
      value=""
    />
     
    <v-text-field-stub
      backgroundcolor=""
      clearicon="$clear"
      color="deep-purple"
      counter="10"
      errorcount="1"
      errormessages=""
      filled="true"
      label="Last name"
      loaderheight="2"
      messages=""
      rules="v => {
          /* istanbul ignore next */cov_15j8hk4u63.f[4]++;
          cov_15j8hk4u63.s[17]++;
          return (/* istanbul ignore next */(cov_15j8hk4u63.b[2][0]++, !!v) || /* istanbul ignore next */(cov_15j8hk4u63.b[2][1]++, 'This field is required')
          );
        }"
      style="min-height: 96px;"
      successmessages=""
      type="text"
      value=""
    />
     
    <v-text-field-stub
      backgroundcolor=""
      clearicon="$clear"
      color="deep-purple"
      counter="10"
      errorcount="1"
      errormessages=""
      filled="true"
      label="Middle name"
      loaderheight="2"
      messages=""
      rules="v => {
          /* istanbul ignore next */cov_15j8hk4u63.f[4]++;
          cov_15j8hk4u63.s[17]++;
          return (/* istanbul ignore next */(cov_15j8hk4u63.b[2][0]++, !!v) || /* istanbul ignore next */(cov_15j8hk4u63.b[2][1]++, 'This field is required')
          );
        }"
      style="min-height: 96px;"
      successmessages=""
      type="text"
      value=""
    />
  </v-form-stub>
   
  <v-divider-stub />
   
  <v-card-actions-stub>
    <v-btn-stub
      activeclass=""
      tag="button"
      text="true"
      type="button"
    >
      Clear
    </v-btn-stub>
     
    <v-spacer-stub />
     
    <v-btn-stub
      activeclass=""
      class="white--text"
      color="deep-purple accent-4"
      depressed="true"
      disabled="true"
      tag="button"
      type="button"
    >
      Submit
    </v-btn-stub>
  </v-card-actions-stub>
</v-card-stub>
`;
